doozerlib/runtime.py

```diff
@@ -399,6 +400,14 @@ def initialize(self, mode='images', clone_distgits=True,
         self.group_dir = self.gitdata.data_dir
         self.group_config = self.get_group_config()
 
+        if self.group_config.assembles.enabled or self.enable_assemblies:
+            if re.match(r'^[a-zA-Z0-9._]+$', self.assembly) is None:
+                raise ValueError('Assembly names may only consistent of alphanumerics, ., and _. ')
```

Comment:
Would also like to disallow `self.assembly` to start or end with `.`:
```suggestion
            if re.fullmatch(r'[\w.]+', self.assembly) is None or self.assembly[0] == '.' or self.assembly[-1] == '.':
                raise ValueError('Assembly names may only consistent of alphanumerics, ., and _, but not start or end with a dot (.).')
```

Reply:
Good call. Updated.